packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -19,8 +19,21 @@ import 'constants.dart';
 import 'icons.dart';
 import 'page_scaffold.dart';
 import 'route.dart';
+import 'search_field.dart';
 import 'theme.dart';
 
+/// Modes that determine when to display the navigation bar's bottom.
+enum NavigationBarBottomMode {
+  /// Enable hiding the bottom in response to scrolling.
+  automatic,
+
+  /// Always display the bottom regardless of the scroll activity.
+  always,
+
+  /// Never display the bottom.
+  none,
+}
+
```

Comment:
I know this is early, but here or somewhere else we should explain how these different modes animate in greater detail.